tests/test_flintrock.py

```diff
@@ -163,4 +163,4 @@ def test_validate_valid_download_source():
 
 def test_validate_invalid_download_source():
     with pytest.raises(Error):
-        validate_download_source("https://www.apache.org/dyn/closer.lua?action=download&filename=hadoop/common/hadoop-2.8.3/hadoop-2.8.3.tar.gz")
+        validate_download_source("https://www.apache.org/dyn/closer.lua?action=download&filename=hadoop/common/hadoop-2.8.3b/hadoop-2.8.3b.tar.gz")
```

Comment:
Oh interesting. The original URL used to be invalid because Apache mirrors don't keep old releases, but I guess now they do?

To make this test more obvious, can we change the version to something like `hadoop-invalid-version`?

Reply:
Done in latest commit